components/automate-ui/src/app/pages/project/list/project-list.component.ts

```diff
@@ -168,46 +201,71 @@ export class ProjectListComponent implements OnInit, OnDestroy {
     this.createProjectForm.reset();
   }
 
-  openConfirmUpdateStartModal() {
+  public openConfirmUpdateStartModal(): void {
     this.confirmApplyStartModalVisible = true;
   }
 
-  closeConfirmApplyStartModal() {
+  private closeConfirmApplyStartModal(): void {
     this.confirmApplyStartModalVisible = false;
   }
 
-  confirmApplyStart() {
-    this.confirmApplyStartModalVisible = false;
+  public confirmApplyStart(): void {
+    this.closeConfirmApplyStartModal();
     this.projects.applyRulesStart();
+    this.applyRulesInProgress = true;
+
+    // Rapid sampling for 3 seconds for more responsive UX.
+    // If the update is still running, the secondary emitter
+    // will check this status at frequent intervals.
+    // Once the update completes, the tertiary emitter
+    // will check this status at INfrequent intervals.
+    // (See getActiveApplyRulesStatus$ and getDormantApplyRulesStatus$.)
+    observableInterval(250).pipe(take(12)) // 12 x 250ms => 3 seconds
+      .subscribe(() => {
+        this.projects.getApplyRulesStatus();
+      });
   }
 
-  cancelApplyStart() {
+  public cancelApplyStart(): void {
     this.closeConfirmApplyStartModal();
   }
 
-  openConfirmUpdateStopModal() {
+  public openConfirmUpdateStopModal(): void {
     this.confirmApplyStopModalVisible = true;
   }
 
-  closeConfirmApplyStopModal() {
+  private closeConfirmApplyStopModal(): void {
     this.confirmApplyStopModalVisible = false;
   }
 
-  confirmApplyStop() {
-    this.confirmApplyStopModalVisible = false;
+  public confirmApplyStop(): void {
+    this.closeConfirmApplyStopModal();
     this.projects.applyRulesStop();
   }
 
-  cancelApplyStop() {
+  public cancelApplyStop(): void {
     this.closeConfirmApplyStopModal();
   }
 
-  getRulesStatus(project: Project): string {
+  public getRulesStatus(project: Project): string {
     switch (project.status) {
       case 'NO_RULES': return 'No rules';
       case 'EDITS_PENDING': return 'Edits pending';
       case 'RULES_APPLIED': return 'Applied';
       default: return '';
     }
   }
+
+  public getProjectStatus(project: Project): string {
```

Comment:
This is the workhorse for populating the "Project Update Status" column and what all the new unit tests exercise.